topo/topo.go

```diff
@@ -229,20 +225,80 @@ func (m *Manager) Load(ctx context.Context) error {
 	return nil
 }
 
-// Topology gets the topology CRDs for the cluster.
-func (m *Manager) Topology(ctx context.Context) ([]topologyv1.Topology, error) {
+// TopologyResources gets the topology CRDs for the cluster.
+func (m *Manager) TopologyResources(ctx context.Context) ([]*topologyv1.Topology, error) {
 	topology, err := m.tClient.Topology(m.proto.Name).List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to get topology CRDs: %v", err)
 	}
-	return topology.Items, nil
+
+	items := make([]*topologyv1.Topology, len(topology.Items))
+	for i := range items {
+		items[i] = &topology.Items[i]
+	}
+
+	return items, nil
 }
 
 // Topology returns the topology protobuf.
 func (m *Manager) TopologyProto() *tpb.Topology {
 	return m.proto
 }
 
+func (m *Manager) TopologySpecs(ctx context.Context) ([]*topologyv1.Topology, error) {
+	nodeSpecs := map[string]*[]*topologyv1.Topology{}
```

Comment:
why is this a `map[string]*[]*...`
you should almost never want a pointer to slice..
the slice is already a pointer to wanting to store a pointer to a pointer is normally a bad idea

Reply:
Thanks for pointing this out. It was initially needed because we were modifying the slice (pointer to slices are inevitable when modifying it). After the refactor, we're just doing RO, so not required anymore !

Reply:
https://github.com/google/kne/pull/138/commits/db2ba56de37628c405180ae609ceb6ee706c4011